src/client/opamArg.ml

```diff
@@ -1525,8 +1528,14 @@ let pin ?(unpin_only=false) () =
     mk_flag ["e";"edit"] "With $(opam pin add), edit the opam file as with \
                           `opam pin edit' after pinning." in
   let kind =
-    let help = "Set the kind of pinning. Must be one of version, \
-                path, git, hg or darcs." in
+    let help =
+      "Sets the kind of pinning. Must be one of $(i,version), $(i,path), \
+       $(i,git), $(i,hg), $(i,darcs) or $(i,auto). \
+       If unset, is inferred from the format of the target, defaulting \
+       to $(i,path). If $(i,auto) or $(i,OPAMPINKINDAUTO) is set, a local \
```

Comment:
`OPAMPINKINDAUTO` should be mentioned [here](https://github.com/ocaml/opam/blob/master/src/client/opamArg.ml#L227).